src/integrationtests/java/com/aws/greengrass/integrationtests/e2e/BaseE2ETestCase.java

```diff
@@ -226,7 +228,11 @@ protected BaseE2ETestCase() throws Exception {
     }
 
     public static void setDefaultRunWithUser(Kernel kernel) {
-        new DeviceConfiguration(kernel).getRunWithDefaultPosixUser().dflt("nobody");
+        String user = "nobody";
+        if (Platform.getInstance() instanceof DarwinPlatform) {
+            user = System.getProperty("user.name");
```

Comment:
Curious why Mac requires an actual valid user

Reply:
`nobody` is a valid user on the mac too - but the default mac sudoers file sets it up so you need to enter the current users password before switching